boot/bootutil/include/bootutil/crypto/ecdh_p256.h

```diff
@@ -43,19 +50,89 @@ static inline int bootutil_ecdh_p256_shared_secret(bootutil_ecdh_p256_context *c
     int rc;
     (void)ctx;
 
-    rc = uECC_valid_public_key(pk, uECC_secp256r1());
+    if (pk[0] != 0x04) {
+        return -1;
+    }
+
+    rc = uECC_valid_public_key(&pk[1], uECC_secp256r1());
     if (rc != 0) {
         return -1;
     }
 
-    rc = uECC_shared_secret(pk, sk, z, uECC_secp256r1());
+    rc = uECC_shared_secret(&pk[1], sk, z, uECC_secp256r1());
     if (rc != TC_CRYPTO_SUCCESS) {
         return -1;
     }
     return 0;
 }
 #endif /* MCUBOOT_USE_TINYCRYPT */
 
+#if defined(MCUBOOT_USE_MBED_TLS)
+typedef struct bootutil_ecdh_p256_context {
+    mbedtls_ecp_group grp;
+    mbedtls_ecp_point P;
+    mbedtls_mpi z;
+    mbedtls_mpi d;
+} bootutil_ecdh_p256_context;
+
+static inline void bootutil_ecdh_p256_init(bootutil_ecdh_p256_context *ctx)
+{
+    mbedtls_mpi_init(&ctx->z);
+    mbedtls_mpi_init(&ctx->d);
+
+    mbedtls_ecp_group_init(&ctx->grp);
+    mbedtls_ecp_point_init(&ctx->P);
+
+    if (mbedtls_ecp_group_load(&ctx->grp, MBEDTLS_ECP_DP_SECP256R1) != 0) {
+        mbedtls_ecp_group_free(&ctx->grp);
+        mbedtls_ecp_point_free(&ctx->P);
+    }
+}
+
+static inline void bootutil_ecdh_p256_drop(bootutil_ecdh_p256_context *ctx)
+{
+    mbedtls_mpi_free(&ctx->d);
+    mbedtls_mpi_free(&ctx->z);
+    mbedtls_ecp_group_free(&ctx->grp);
+    mbedtls_ecp_point_free(&ctx->P);
+}
+
+static inline int bootutil_ecdh_p256_shared_secret(bootutil_ecdh_p256_context *ctx, const uint8_t *pk, const uint8_t *sk, uint8_t *z)
+{
+    int rc;
+
+    rc = mbedtls_ecp_point_read_binary(&ctx->grp,
+                                       &ctx->P,
+                                       pk,
+                                       EC_TAG_OFFSET);
```

Comment:
`EC_TAG_OFFSET` has no meaning here, this should be called `EC_PUBK_LEN`, or `EC256_PUBK_LEN`.